CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
+This document tries to document them as much as possible.
+
+A *few* of these rules are checked automatically using Scalastyle, but most of them are too complex to teach to an automated tool.
+
+In some cases, [@sjrd](https://github.com/sjrd) will just say something else on a code review.
+His opinion on a direct piece of code always takes precedence over this document.
+Feel free to point out the conflict, though: the document can be improved.
```

Comment:
@yawnt while I again agree that this can be formulated friendlier, I must point out that this is not *our* (for whatever definition of our) coding style, but @sjrd's. And that is a good thing. It is **much** more important to have a consistent style than everyone being happy with his/her own little preferences.

I agree that this can sound harsh or maybe even arrogant, maybe because it evokes the fear of having to deal with an unreasonable person enforcing their dictatorship. This whole thing of course only works if the person in power is the exact opposite (for which, IMHO, @sjrd easily qualifies). But how can we express that without an essay?? :)